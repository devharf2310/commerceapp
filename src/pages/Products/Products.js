import React, { useEffect, useState } from "react";
import ProductsList from "../../components/ProductsList/ProductsList";
import { getProducts, getTotalAmountProducts } from "../../utils/DataUtils";
import { CircularProgress, Grid } from "react-md";
import './Products.scss';

const Products = ({match}) => {
  const [products, setProducts] = useState([]);
  const [progress, setProgress] = useState(null);

  useEffect(() => {
    fetchProducts();
  }, []);

  async function fetchProducts() {
    setProgress(undefined);
    await getProducts(setProducts, match.params.category);
    setProgress(null);
  }

  return (
    <Grid>
      {progress !== null && <CircularProgress value={progress}/>}
      {progress === null &&
      <p>
          Showing <span className="span-quantity-products">{products.length}</span> -
          Hidden <span className="span-quantity-products">{getTotalAmountProducts() - products.length}</span>
      </p>}
      <ProductsList products={products} />
    </Grid>
  );
};

export default Products;


