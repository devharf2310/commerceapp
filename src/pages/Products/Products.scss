.span-quantity-products{
    font-weight: bolder;
}
