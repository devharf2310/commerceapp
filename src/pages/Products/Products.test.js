import React from 'react'
import renderer from 'react-test-renderer'
import Products from './Products';

describe('Products snapshot tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Products/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
