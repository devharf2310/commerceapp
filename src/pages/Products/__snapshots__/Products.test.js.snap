// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products snapshot tests renders correctly 1`] = `
<div
  className="md-grid"
>
  <p>
    Showing 
    <span
      className="span-quantity-products"
    >
      0
    </span>
     - Hidden 
    <span
      className="span-quantity-products"
    >
      35
    </span>
  </p>
</div>
`;
